core/bin/external_node/src/config/mod.rs

```diff
@@ -471,16 +471,20 @@ impl PostgresConfig {
     }
 }
 
-pub(crate) fn read_consensus_config() -> anyhow::Result<consensus::FetcherConfig> {
-    let path = std::env::var("EN_CONSENSUS_CONFIG_PATH")
-        .context("EN_CONSENSUS_CONFIG_PATH env variable is not set")?;
+pub(crate) fn read_consensus_secrets() -> anyhow::Result<Option<consensus::Secrets>> {
+    let Ok(path) = std::env::var("EN_CONSENSUS_SECRETS_PATH") else {
+        return Ok(None);
+    };
     let cfg = std::fs::read_to_string(&path).context(path)?;
-    let cfg: consensus::config::Config =
-        consensus::config::decode_json(&cfg).context("failed decoding JSON")?;
-    let node_key: node::SecretKey = consensus::config::read_secret("EN_CONSENSUS_NODE_KEY")?;
-    Ok(consensus::FetcherConfig {
-        executor: cfg.executor_config(node_key),
-    })
+    Ok(Some(decode_json(&cfg).context("failed decoding JSON")?))
+}
+
+pub(crate) fn read_consensus_config() -> anyhow::Result<Option<consensus::Config>> {
+    let Ok(path) = std::env::var("EN_CONSENSUS_CONFIG_PATH") else {
+        return Ok(None);
+    };
+    let cfg = std::fs::read_to_string(&path).context(path)?;
+    Ok(Some(decode_json(&cfg).context("failed decoding JSON")?))
```

Comment:
why json? 
if we use yaml in other places?

Reply:
nobody protested so far. I can change it to yaml.

Reply:
done